crates/oxc_transformer/src/context.rs

```diff
@@ -10,9 +10,9 @@ use oxc_ast::AstBuilder;
 use oxc_diagnostics::Error;
 use oxc_semantic::Semantic;
 
-use crate::{helpers::module_imports::ModuleImports, TransformOptions};
+use crate::{helpers::module_imports::ModuleImports, naming::TransformerNaming, TransformOptions};
 
-pub type Ctx<'a> = Rc<TransformCtx<'a>>;
```

Comment:
`TransformerNaming` needs to mutation.

In a previous implementation. All the `TransformerCtx` fields that cannot be copied are wrapped in `Rc`
https://github.com/oxc-project/oxc/blob/31ed532b790691b843182768351477dd26428b94/crates/oxc_transformer/src/context.rs#L15-L22

Reply:
This is a big change.

It probably makes more sense to use interior mutability in `TransformerNaming`, instead of adding all these borrow calls everywhere.

Reply:
> It probably makes more sense to use interior mutability in `TransformerNaming`, instead of adding all these borrow calls everywhere.

Yes, I agree. When we finally consider this PR, I'll refactor it.